src/lib.rs

```diff
@@ -147,6 +147,7 @@ mod log {
 
 mod dns_parser;
 mod error;
+#[allow(clippy::cognitive_complexity)]
```

Comment:
Since I suggest we don't add `clippy::nursery` in CI,  we don't need this line either.